mewarpx/mewarpx/mwxrun.py

```diff
@@ -103,23 +104,41 @@ def init_timestep(self, V_anode, DT=None, CFL_factor=None, V_grid=5):
         return self.dt
 
 
-    def init_run(self):
+    def init_run(self, restart=None, checkpoint_dir="diags",
+                 checkpoint_prefix="checkpoint", additional_steps=None):
         if self.initialized:
             raise RuntimeError(
                 "Attempted to initialize the mwxrun class multiple times.")
-
-        self.simulation.initialize_inputs()
-        self.simulation.initialize_warpx()
-
-        self.me = _libwarpx.libwarpx.warpx_getMyProc()
-        self.n_procs = _libwarpx.libwarpx.warpx_getNProcs()
-
-        # A level is needed for many things like level number. For now I'm
-        # statically setting the default level here. I'm not sure of pitfalls
-        # or how to handle it more generally yet.
-        self.lev = _libwarpx.libwarpx.warpx_finestLevel()
-
-        self.initialized = True
+        try:
+            if restart != False:
+                restart = init_restart_util.run_restart(
+                    checkpoint_dir, checkpoint_prefix,
+                    bool(restart), additional_steps
+                )
+
+            self.simulation.initialize_inputs()
+            self.simulation.initialize_warpx()
+
+            self.me = _libwarpx.libwarpx.warpx_getMyProc()
+            self.n_procs = _libwarpx.libwarpx.warpx_getNProcs()
+
+            # A level is needed for many things like level number. For now I'm
+            # statically setting the default level here. I'm not sure of pitfalls
+            # or how to handle it more generally yet.
+            self.lev = _libwarpx.libwarpx.warpx_finestLevel()
+
+            self.initialized = True
+
+        except RuntimeError:
+            if not restart:
```

Comment:
Wouldn't `restart` be `True` here if `init_restart_util.run_restart()` found a checkpoint file to restart from? So then this line should be the following, no?
```suggestion
            if restart:
```

Reply:
Ah yep